drivers/extend/gpio_redir.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2018 Acutam Automation, LLC
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_extend_gpio
+ *
+ * @{
+ *
+ * @file
+ * @brief       GPIO extension redirection functions
+ *
+ * @author      Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * @}
+ */
+
+#include "periph/gpio.h"
+#include "extend/gpio.h"
+#include "gpio_ext_conf.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+gpio_ext_t *gpio_ext_entry(gpio_t gpio)
+{
+    if (gpio == GPIO_UNDEF) {
+        return NULL;
+    }
+
+    gpio_t devnum = gpio_ext_dev(gpio);
+
+    DEBUG("[gpio_ext_entry] list entry is %u\n", devnum);
+
+    /* device is greater than number of listed entries */
+    if (devnum > (sizeof(gpio_ext_list) / sizeof(gpio_ext_list[0]))) {
```

Comment:
This has to be
```
if (devnum >= (sizeof(gpio_ext_list) / sizeof(gpio_ext_list[0]))) {
```
since the valid entry indices go from 0 ... number of entries - 1. For example, when you have only one entry and you try to access to `GPIO_EXT_PIN(1,0)`, `gpio_ext_dev` gives 1 which is equal to the number of entries and leads to a crash at the moment.

Reply:
Thanks for the catch. This is a holdover from when devnum was defined slightly differently. Fixed.